retros/README.md

```diff
@@ -0,0 +1,12 @@
+# Retros
+
+We have a retro at least once a month. The retro deputy is responsible for scheduling and facilitating the retro meeting. The retro deputy is responsible for finding a retro deputy for the following month and reminding them about retro deputy duties.
```

Comment:
As per our discussion at the handbook meeting: can you add a bit here about WHY (i.e. the goal of retros and why we have them at this cadence) so the next person to see this doc knows why it's here and how best to change it?

Reply:
@maiamcc : Can you please review again?